### PR TITLE
fix: BasePath not included in AWS::ApiGateway::BasePathMapping when defined as ["/"]

### PR DESCRIPTION
**Pull request by mbfreder**
### Issue #, if available
The "AWS::Serverless::Api" SAM resource type creates an "AWS::ApiGateway::BasePathMapping" resource when processed.
When the 'BasePath' sub-property of this resource type is defined as ["/"] , the "AWS::ApiGateway::BasePathMapping" resource in the processed template does NOT include the 'BasePath' property.

### Description of changes
transform ["/"] into "" in the processed template.

### Description of how you validated changes
Updated unit tests to cover the edge case

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vicheey** [comment]
For context, this is the previous PR that contain change that this current PR try to address : https://github.com/aws/serverless-application-model/pull/3719

**Comment by valerena** [comment]
It's probably better to add **new** test files instead of modifying the existing ones. It's always good to keep old tests relatively steady, so we can confirm that we're not breaking any of the old cases that were already working fine.

File: samtranslator/model/api/api_generator.py [hunk 1]
```diff
@@ -805,8 +805,7 @@ def _create_basepath_mapping(
         basepath_mapping.DomainName = ref(api_domain_name)
         basepath_mapping.RestApiId = ref(rest_api.logical_id)
         basepath_mapping.Stage = ref(rest_api.logical_id + ".Stage")
-        if basepath:
-            basepath_mapping.BasePath = basepath
+        basepath_mapping.BasePath = basepath
```
Comment (valerena) [inline samtranslator/model/api/api_generator.py:808]: I think we still need this if, because we use this method passing `None` as `basepath` (and `basepath` is optional), so we should probably actually update this to `if basepath is not None:` instead of removing the `if`. 

I think that's a different edge case that we might not be covering in the test cases either, which is the first branch of this `if`: https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L546-L548

Comment (vicheey) [inline samtranslator/model/api/api_generator.py:808]:
```suggestion
        # Skip checking the basepath; We need to set "BasePath": "" if no basepath is provided to satisfy CFN. 
        basepath_mapping.BasePath = basepath
```

**Review by valerena — APPROVED** [review]

**Review by vicheey — APPROVED** [review]
